### PR TITLE
busctl: Misc cleanups and a fix

### PR DESCRIPTION
**Pull request by namhyung**
This is a couple of cleanups and a fix on sd-bus/busctl code.  It tries to cleanup getpeersec() and member_compare_func().  It also adds description of --size option in busctl and fixes a bug in the option handling code.

### CODE REVIEW

**Comment by zonque** [comment]
> Anyway, this patch is just for cleanup and I think it'd help readability of the code a little bit. If you think it's not needed I can drop this patch then. Thanks for your review.

Jup, let's drop that one. The others look good, except for the missing tests. Thanks!

File: src/basic/util.c [hunk 1]
```diff
@@ -128,6 +128,20 @@ bool streq_ptr(const char *a, const char *b) {
         return false;
 }
 
+int strcmp_ptr(const char *a, const char *b) {
+
+        /* Like strcmp(), but tries to make sense of NULL pointers */
+
+        if (a && b)
+                return strcmp(a, b);
+
```
Comment (zonque) [inline src/basic/util.c:137]: Nit pick: stray newline.

File: src/basic/util.c [hunk 2]
```diff
@@ -128,6 +128,20 @@ bool streq_ptr(const char *a, const char *b) {
...
+        if (!a && b)
+                return -1;
+        if (a && !b)
+                return 1;
```
Comment (zonque) [inline src/basic/util.c:141]: Rather add one here :)

File: src/basic/util.c (hunk 1 above)
Comment (namhyung) [inline src/basic/util.c:137]: Oh, it's copied from streq_ptr() :)

File: src/basic/util.c (hunk 2 above)
Comment (namhyung) [inline src/basic/util.c:141]: So, do you want this?

```
/* Like strcmp(), but tries to make sense of NULL pointers */
if (a && b)
    return strcmp(a, b);

if (!a && b)
    return -1;

if (a && !b)
    return 1;
```

Comment (zonque) [inline src/basic/util.c:141]: Yes, looks good!

**Comment by poettering** [comment]
Looks good in general! Thanks!

**Comment by namhyung** [comment]
I'm not familiar enough with github, and I don't know what is a good way to update this PR after finishing rework.  Is it okay if I 'git push --force' to the same branch?

**Comment by zonque** [comment]
> I'm not familiar enough with github, and I don't know what is a good way to update this PR after finishing rework. Is it okay if I 'git push --force' to the same branch?

That's one way, but we usually refrain from that. Just close this one, and open another PR with the new patches. Thanks!

**Comment by namhyung** [comment]
> That's one way, but we usually refrain from that. Just close this one, and open another PR with the new patches. Thanks!

Got it.  Thanks!

**Pull request closed without merging**